src/platform/android/CommissionableDataProviderImpl.cpp

```diff
@@ -0,0 +1,217 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "CommissionableDataProviderImpl.h"
+
+#include <cstdint>
+#include <string.h>
+
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/support/Base64.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/JniTypeWrappers.h>
+#include <lib/support/Span.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/CHIPDeviceConfig.h>
+
+using namespace chip;
+using namespace chip::Crypto;
+
+namespace {
+
+CHIP_ERROR GeneratePaseSalt(std::vector<uint8_t> & spake2pSaltVector)
+{
+    constexpr size_t kSaltLen = kSpake2p_Max_PBKDF_Salt_Length;
+    spake2pSaltVector.resize(kSaltLen);
+    return DRBG_get_bytes(spake2pSaltVector.data(), spake2pSaltVector.size());
+}
+
+} // namespace
+
+CommissionableDataProviderImpl CommissionableDataProviderImpl::sInstance;
+
+CHIP_ERROR CommissionableDataProviderImpl::Update(JNIEnv * env, jstring spake2pVerifierBase64, jstring Spake2pSaltBase64,
+                                                  jint spake2pIterationCount, jlong setupPasscode, jint discriminator)
+{
+    if (!mIsUpdate)
```

Comment:
any reason this is done before argument validation? If so, please add a comment explaining, otherwise move it after argument validation.

I would expect argument validation to always be the first thing to be done before any other actions in a function.

Reply:
fixed